### PR TITLE
Sort imports, add check in CI

### PR DESCRIPTION
**Pull request by pawamoy**
For issue #1625

First commit adds the configuration for isort in setup.cfg, the tox env in tox.ini, and the job in travis config.

Next commit will actually fix the imports, to see if CI passes.

### CODE REVIEW

**Comment by pawamoy** [comment]
I think #1618 should be done first, to avoid ordering imports now and re-ordering them after #1628.

**Comment by pawamoy** [comment]
Job correctly failing, pushing the fix :slightly_smiling_face:

**Comment by ramnes** [comment]
Yisss! I don't like the 120 line length though, but I guess we'll discuss this in the issue about code formatting.

**Comment by pawamoy** [comment]
I can change it, no problem. Thing is, flake8 is configured with a line length of 120 so I used the same length :)

**Comment by ramnes** [comment]
Oh! If flake8 is already configured to 120, then don't mind me.

**Comment by tych0** [comment]
I think I just did that to avoid having to reformat everything. I think we
can definitely have a discussion about shrinking it.

**Comment by pawamoy** [comment]
- should I put the isort command inside of the pep8 tox env instead of its own?
- what line length?

**Comment by tych0** [comment]
Putting it in pep8 is fine. For imports, seems like we should be able to get away with 80 characters pretty easily for a line length.

**Comment by ramnes** [comment]
You might want to use https://github.com/gforcada/flake8-isort

**Comment by pawamoy** [comment]
Alright I'll update this. I should definitely take a look at all those flake8 plugins :smile:

**Comment by pawamoy** [comment]
Some things detected by `flake8-isort` are not fixed by running `isort` itself with the same configuration.

```
./libqtile/widget/battery.py:40:1: I001 isort found an import in the wrong position
./libqtile/widget/battery.py:47:1: I003 isort expected 1 blank line in imports, found 0
./libqtile/widget/battery.py:47:1: I003 isort expected 1 blank line in imports, found 0
./libqtile/widget/generic_poll_text.py:21:1: E303 too many blank lines (3)
```

I fixed them manually.

The first three were due to an import kept apart because of a `noqa` comment.

```python
from typing import Dict  # noqa: F401; noqa: F401
from typing import Any, List, NamedTuple, Optional, Tuple
```

Fixed by

```python
from typing import Any, Dict, List, NamedTuple, Optional, Tuple
```

Not sure what the F401 warning is about.

File: libqtile/images.py [hunk 1]
```diff
@@ -8,6 +8,11 @@
 # The above copyright notice and this permission notice shall be included in
 # all copies or substantial portions of the Software.
 
+import io
+import os
+import re
+from collections import OrderedDict, defaultdict, namedtuple
+
```
Comment (pawamoy) [inline libqtile/images.py:15]: Hmm, this doesn't seem right. I think we should concatenate both comments section to the imports are not inserted between the two.

File: test/layouts/test_bsp.py [hunk 2]
```diff
@@ -17,12 +17,13 @@
 # OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
 # SOFTWARE.
 
+from test.conftest import no_xinerama
+from test.layouts.layout_utils import assert_focus_path, assert_focused
+
```
Comment (pawamoy) [inline test/layouts/test_bsp.py:22]: I think we should mark `test` as third-party in isort's config. Will fix it.

**Comment by pawamoy** [comment]
I'm not sure why but `flake8` does not detect unused imports in the tests, for example `import libqtile.bar` and `import libqtile.scratchpad` in `test_scratchpad.py`.

**Comment by tych0** [comment]
This all looks great to me, thank you for doing it!

**Pull request merged**